packages/flutter_tools/gradle/src/main/kotlin/FlutterPluginUtils.kt

```diff
@@ -1,26 +1,34 @@
 package com.flutter.gradle
 
+import com.android.build.gradle.AbstractAppExtension
+import com.android.build.gradle.BaseExtension
 import com.android.builder.model.BuildType
 import groovy.lang.Closure
 import org.gradle.api.GradleException
+import org.gradle.api.JavaVersion
 import org.gradle.api.Project
 import org.gradle.api.Task
 import org.gradle.api.UnknownTaskException
 import org.gradle.api.logging.Logger
 import java.io.File
+import java.nio.charset.StandardCharsets
+import java.util.Properties
 
 /**
  * A collection of static utility functions used by the Flutter Gradle Plugin.
  */
 object FlutterPluginUtils {
```

Comment:
This pr is complicated enough so this comment is not blocking but ... 
I think these methods should longer term go into a file like `FlutterTaskAddingUtils` and keep `FlutterPluginUtils` focused on code that needs to be shared across tasks. 

This is the danger of any *Utils style file it does not articulate the scope of what should be contained.